codegenerator/cli/src/config_parsing/entity_parsing.rs

```diff
@@ -1851,4 +2030,333 @@ type TestEntity {
+            true, // is_array
+        );
+
+        // BigDecimal fields
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimal")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(80),
+            Some(5),
+            false, // is_required
+            false, // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimalRequired")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(81),
+            Some(5),
+            true,  // is_required
+            false, // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimalArray")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(82),
+            Some(5),
+            false, // is_required
+            true,  // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimalArrayRequired")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(83),
+            Some(5),
+            true, // is_required
+            true, // is_array
+        );
+
+        // exampleBigDecimalOtherOrder
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimalOtherOrder")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(84),
+            Some(6),
+            true,  // is_required
+            false, // is_array
+        );
+    }
+
+    #[test]
+    fn test_error_case_numeric_on_bigint() {
+        let schema_str = r#"
+        type Entity {
+            id: ID!
+            exampleBigIntWrongDirective: BigInt @numeric(precision: 76, scale: 5)
+        }
+        "#;
+
+        let gql_doc = setup_document(schema_str).expect("Failed to parse schema");
+        let result = Schema::from_document(gql_doc);
+
+        assert!(result.is_err());
+        let err_message = format!("{:?}", result.unwrap_err());
+        assert!(err_message.contains(
+            "The numeric directive on a field is only suitable for BigDecimal scalar type."
+        ));
+    }
+
+    #[test]
+    fn test_error_case_decimal_precision_on_bigdecimal() {
+        let schema_str = r#"
+        type Entity {
+            id: ID!
+            exampleBigIntWrongDirective: BigDecimal @precision(precision: 76)
+        }
+        "#;
+
+        let gql_doc = setup_document(schema_str).expect("Failed to parse schema");
+        let result = Schema::from_document(gql_doc);
+
+        assert!(result.is_err());
+        let err_message = format!("{:?}", result.unwrap_err());
+        assert!(err_message.contains(
+            "The precision directive on a field is only suitable for BigInt scalar type."
+        ));
```

Comment:
I think we should really start leaning on this error for these kinds of case testing. String matching is not super nice for error testing.   Especially since you wan to be able to update messages easily.

Reply:
I agree that it does add some brittleness. Can you explain what you mean by leaning on this area? I feel like you didn't complete that sentence fully.